WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift

```diff
@@ -494,10 +495,41 @@ class ReaderDetailCoordinator {
     /// Given a URL presents it in a new Reader detail screen
     ///
     private func presentReaderDetail(_ url: URL) {
-        let readerDetail = ReaderDetailViewController.controllerWithPostURL(url)
+
+        // If the user tapped the link to the original post, use the original post's info to display reader detail.
+        // The API endpoint used by controllerWithPostID returns subscription flags for the post.
+        // The API endpoint used by controllerWithPostURL does return this information.
+        // These flags needed to display the `Follow conversation by email` option.
+        // So if we can call controllerWithPostID, do so. Otherwise, fallback to controllerWithPostURL.
+        // Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/17158
+
+        let readerDetail: ReaderDetailViewController = {
+            if let post = post,
+               selectedUrlIsCrossPost(url) {
+                return ReaderDetailViewController.controllerWithPostID(post.crossPostMeta.postID, siteID: post.crossPostMeta.siteID)
+            }
+
+            return ReaderDetailViewController.controllerWithPostURL(url)
+        }()
+
         viewController?.navigationController?.pushViewController(readerDetail, animated: true)
     }
 
+    private func selectedUrlIsCrossPost(_ url: URL) -> Bool {
```

Comment:
I'll be "that guy" and ask if we should add a unit test for this?

Reply:
Oh I suppose we could. 😄 I can do that in a follow up PR if you don't mind.

Reply:
wfm!